compliance_suite/functions/client.py

```diff
@@ -59,6 +59,7 @@ def send_request(
         for key in uri_params.keys():
             endpoint = endpoint.replace(f"{{{key}}}", uri_params[key])
 
+        version = "v" + version[0]  # Convert SemVer into Major API version
```

Comment:
And what happens if you use this on an API whose major version consists of more than one digit?

Also, as I see that you are using this version parsing code in more than one location, perhaps it's better to define a small function, then call that. That would also make it easier to unit test and fix in case there is some other issue we didn't think about in the parser.

On another note: I find it a bit strange that you use a class here, but basically don't make much use of it in terms of passing data around. Instead, some of the methods have a ton of arguments (and identical ones). I think this could be refactored.

It might help you get used to this by using a strict linter like Pylint, which will nag when you use too many arguments etc. It's fair though, because nobody likes to use functions/methods with a ton of options if it can be avoided.

No need to change this now, but perhaps you can add this as a low-priority refactoring issue.

Reply:
Sorry, I should have taken the split fn. I have added it now.

I have refactored the code to remove redundancy and add more intuitive code.